README.md

```diff
@@ -162,31 +163,47 @@ test[0].bar.x = 2;
 
 ### Omit
 
+Removed in `v3`.
+
 NOTE: Builtin `Omit` became part of TypeScript 3.5
 
+### DeepOmit
+
+Recursively omit deep properties according to key names.
+
+Here is the `Teacher` interface.
+
 ```typescript
-type ComplexObject = {
-  simple: number;
-  nested: {
-    a: string;
-    array: [{ bar: number }];
-  };
-};
+interface Teacher {
+  name: string,
+  gender: string,
+  students: {name: string, score: number}[]
+}
+```
 
-type SimplifiedComplexObject = Omit<ComplexObject, "nested">;
+Now suppose you want to omit `gender` property of `Teacher`, and `score` property of `students`. You can simple define a filter type and then use `DeepOmit`.
 
-// Result:
-// {
-//  simple: number
-// }
+In the filter type, the properties to be omitted completely should be defined as `true`. For the properties you want to partially omit, you should recursively define the sub-properties to be omitted.
+
+```typescript
+interface Filter {
```

Comment:
what do you think about inlining filter? I guess this is what users would rather do because Filter type is not useful anywhere else.

Reply:
+1, in the example usage it's enough to inline it

Reply:
I explicitly defined the Filter type here to make it clearer. However, if you think inlining is better, I can use inlining filter in the README instead.